source/common/http/header_utility.cc

```diff
@@ -180,6 +180,33 @@ bool HeaderUtility::isEnvoyInternalRequest(const RequestHeaderMap& headers) {
          internal_request_header->value() == Headers::get().EnvoyInternalRequestValues.True;
 }
 
+void HeaderUtility::stripPortFromHost(RequestHeaderMap& headers, uint32_t listener_port) {
+  const auto original_host = headers.Host()->value().getStringView();
+  const absl::string_view::size_type port_start = original_host.rfind(':');
+  if (port_start == absl::string_view::npos) {
+    return;
+  }
+  // according to RFC3986 v6 address is always enclosed in "[]". section 3.2.2
+  const auto v6_end_index = original_host.rfind("]");
+  if (v6_end_index == absl::string_view::npos || v6_end_index < port_start) {
+    if ((port_start + 1) > original_host.size()) {
+      return;
```

Comment:
Arguably this is taken care of by the atoi checks (for ipv6 without host, the data after the last : will not be an integer).  If you prefer having the correct index for clarity it might be worth making a shared utility with processRequestHost (I knew we normalized port somewhere else!) for finding port index, where we either rfind : or ]:  as I think it's a bit easier to read

Reply:
> Arguably this is taken care of by the atoi checks (for ipv6 without host, the data after the last : will not be an integer).

from my understanding  `v6_end_index < port_start` should take care of this (for something like [fc00::1] index of ':' would be smaller than v6_end_part (aka index of ']'))

> If you prefer having the correct index for clarity it might be worth making a shared utility with processRequestHost (I >knew we normalized port somewhere else!) for finding port index, where we either rfind : or ]: as I think it's a bit easier to > read

let me check what processRequestHost is doing. if i could share logic for v6 parsing w/ it - less work to do ;)